src/rtnl/link/nlas/link_xdp.rs

```diff
@@ -0,0 +1,322 @@
+        let nlas = NlasIterator::new(buf.into_inner());
+        for nla in nlas {
+            let nla = nla?;
+            match nla.kind() as u32 {
+                IFLA_XDP_UNSPEC => res.push(Xdp::Unspec(nla.value().to_vec())),
+                IFLA_XDP_FD => res.push(Xdp::Fd(
+                    parse_i32(nla.value())
+                        .context("invalid IFLA_XDP_FD value")?,
+                )),
+                IFLA_XDP_ATTACHED => res.push(Xdp::Attached(
+                    XdpAttached::try_from(nla.value()[0])
+                        .context("invalid IFLA_XDP_ATTACHED value")?,
+                )),
+                IFLA_XDP_FLAGS => res.push(Xdp::Flags(
+                    parse_u32(nla.value())
+                        .context("invalid IFLA_XDP_FLAGS value")?,
+                )),
+                IFLA_XDP_PROG_ID => res.push(Xdp::ProgId(
+                    parse_u32(nla.value())
+                        .context("invalid IFLA_XDP_PROG_ID value")?,
+                )),
+                IFLA_XDP_DRV_PROG_ID => res.push(Xdp::DrvProgId(
+                    parse_u32(nla.value())
+                        .context("invalid IFLA_XDP_PROG_ID value")?,
+                )),
+                IFLA_XDP_SKB_PROG_ID => res.push(Xdp::SkbProgId(
+                    parse_u32(nla.value())
+                        .context("invalid IFLA_XDP_PROG_ID value")?,
+                )),
+                IFLA_XDP_HW_PROG_ID => res.push(Xdp::HwProgId(
+                    parse_u32(nla.value())
+                        .context("invalid IFLA_XDP_PROG_ID value")?,
+                )),
+                IFLA_XDP_EXPECTED_FD => res.push(Xdp::ExpectedFd(
+                    parse_u32(nla.value())
+                        .context("invalid IFLA_XDP_PROG_ID value")?,
+                )),
+                _ => {
+                    return Err(
+                        format!("unknown NLA type {}", nla.kind()).into()
+                    )
+                }
+            }
+        }
+        Ok(VecXdp(res))
+    }
+}
+
+#[repr(u8)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum XdpAttached {
+    /// XDP_ATTACHED_NONE
+    None = XDP_ATTACHED_NONE,
+    /// XDP_ATTACHED_DRV
+    Driver = XDP_ATTACHED_DRV,
+    /// XDP_ATTACHED_SKB
+    SocketBuffer = XDP_ATTACHED_SKB,
+    /// XDP_ATTACHED_HW
+    Hardware = XDP_ATTACHED_HW,
+    /// XDP_ATTACHED_MULTI
+    Multiple = XDP_ATTACHED_MULTI,
+}
+
+impl TryFrom<u8> for XdpAttached {
+    type Error = DecodeError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            XDP_ATTACHED_NONE => Ok(XdpAttached::None),
+            XDP_ATTACHED_DRV => Ok(XdpAttached::Driver),
+            XDP_ATTACHED_SKB => Ok(XdpAttached::SocketBuffer),
+            XDP_ATTACHED_HW => Ok(XdpAttached::Hardware),
+            XDP_ATTACHED_MULTI => Ok(XdpAttached::Multiple),
+            _ => Err(format!("unknown xdp attached type {}", value).into()),
```

Comment:
In other part of this project, we add `Other(u8)` to hold the unknown date instead of raise error. Your code will cause tools using this crate not able to run on newer kernel.